# Analyzing RADSeq Data
This repository contains my work from the Bioinformatics class FISH 546 Autumn 2018.  
  

## Project Description
The goal of this project is to process data from Restriction-site Associated DNA Sequencing (RADSeq) done on individuals from the family Liparidae, the snailfishes, in order to eventually use that data for a phylogenetic tree.  
This data set was generated by Sam Ghods, Ingrid Spies, and Jay Orr in a collaboration between the University of Washington Fish Collection and the Alaska Fisheries Science Center.  
## Question:
How do I go from raw RADSeq sequencing data to stacks of loci that could be used for constructing a phylogeny?  
**Bonus Question**  
How do I use the command line to automate downloading and renaming CT scanning files to save time and remove user error?
### RADSeq Data Processing:
1. Demultiplex and trim sequences using Stacks.
2. Filter by read number and cap samples with too many reads.  
3. Determine number of loci within each individual.   
4. Create catalog of loci from all individuals to align against (becasue no genome to align against)
5. Align loci to catalog.

**Bonus CT Processing:**
1. For single file, curl from website, unzip, scrub Info.txt file for rename data, rename .log file, .zip file, and all images in .zip file.  
2. Loop so that it takes input from a .txt file with a list of numbers you want to download.  
3. Make sure it is compatible with python batch uploading script.  

### Project Timeline
**Completed**  
**Week 3**: Organize repository and gain access to raw data   
**Week 4**: Have data downloaded, inspected using FastQC, and trimmed using Trim Galore.     
**Week 5**: Install Stacks, very common pipeline for analyzing RADSeq data  
*Note installing Stacks was extremely time intensive, see [InstallingStacks](https://github.com/fish546-2018/Jenny-Snailfish/blob/master/progress/InstallingStacks.md)*   
**Week 6**: Demultiplex data, and trim barcodes. Step completed but results didn't save due to Virtual Machine running out of space.  
*Pivot to writing script for automating CT data downloading*  
**Week 7**: Resolve issues with virtual machine running out of space, by deleting all non-essential things and removing shared folders. Accessed data from external harddrive instead.  
*Finish script for CT downloading, make it work with already written batch uploading script*  
**Week 8**: Rerun `process_radtags` now that VM issues are resolved.    
**Week 9**: Run `ustacks` to determine loci within each individual. Run `cstacks` to create a catalog to align against.  
**Week 10**: Troubleshoot issues with `cstacks`, align against catalog using `sstacks`.  

## Repository Structure:
Detailed repository structure can be found [here](https://github.com/fish546-2018/Jenny-Snailfish/blob/master/RepoOrg.md)

## Programs Used for Data Processing
- FastQC 
	- requires java
- Trim Galore
	- requires Cutadapt (which requires Microsoft C++)
- Stacks
