# Liparidae phylogeny based on RADSeq Data
This repository contains my work from the Bioinformatics class FISH 546 Autumn 2018.  
  

## Project Description
This project will be working with RADSeq data from snailfishes. This data set was generated by Sam Ghods, Ingrid Spies, and Jay Orr in a collaboration between the University of Washington Fish Collection and the Alaska Fisheries Science Center.  
## My goal is to go from the raw RADSeq data to a phylogeny based on that data. 
### Objectives:
1. Determine number of usable loci for all species
2. Determine if more loci are usable for groups of more closely related species
3. Construct a phylogeny of all species
4. Construct a phylogeny of groups of closely related species that aren't fully resolved on the full tree.
### Project Timeline
Week 3: Organize repository and gain access to data  
Week 4: Have data downloaded, inspected, and trim low quality bases. Install programs needed for further analysis/phylogeny construction.  
Week 5 goals: Become familiar with new programs and begin running basic analyses.
## Directories:  
- xFISH546
	- general class files including tutorials and class notes
	- blast-tutorial
		- files created from conducting the blast tutorial in GitBash
		- contains text file of command line coding
		- contains Jupyter notebook for running the tutorial
		- contains subdirectories: analyses for blast analysis, db for data downloads, and query for data queries.
	- notebooks
		- contains Jupyter notebooks about general troubleshooting done for making Bash and Windows get along.
	-  notes
		-  notes from each week of class
	-  screenshots
		-  screenshots taken for posting on class issues
	-  shell-novice
		-  files from the bash shell tutorial
		-  two text files with commands from the tutorial
		-  data -> file downloaded for use with the bash tutorial
-  Data
	-  contains raw data file and checksum for it
	-  *original data fastq too big for GitHub*
-  Analyses
	-  contains files created for analysis of data
	-  currently contains html and zip output of FastQC
-  Notebooks
	-  contains Jupyter notebooks documenting steps in the process
	-  
## Programs Used
- FastQC 
	- requires java
- Trim Galore
	- requires Cutadapt (which requires Microsoft C++)
- Stacks
