# Liparidae phylogeny based on RADSeq Data
This repository contains my work from the Bioinformatics class FISH 546 Autumn 2018.  
  

## Project Description
The goal of this project is make a maximum likelihood phylogentic tree based of data from Restriction-site Associated DNA Sequencing (RADSeq) done on individuals from the family Liparidae, the snailfishes.  
This data set was generated by Sam Ghods, Ingrid Spies, and Jay Orr in a collaboration between the University of Washington Fish Collection and the Alaska Fisheries Science Center.  
## Question:
For snailfishes, how does a phylogeny based on many single nucleotide polymorphisms from across the genome compare to a phylogeny generated using Cytochrome Oxidase I sequences? Do they unconver the same or different relationships?
### Data Processing:
1. Demultiplex and trim sequences using Stacks, then filter by read number.
2. Determine number of usable loci for all species using Stacks.
3. Determine if more loci are usable for groups of more closely related species.
4. Construct a phylogeny of all species based on extracted SNPs
5. Compare differences between RADSeq phylogeny and already generated COI phylogeny.

### Project Timeline
**Completed**  
**Week 3**: Organize repository and gain access to raw data **DONE**  
**Week 4**: Have data downloaded, inspected using FastQC, and trimmed using Trim Galore. **DONE**     
**Week 5**: Install Stacks, very common pipeline for analyzing RADSeq data **DONE**  
*Note installing Stacks was extremely time intensive, see [InstallingStacks](https://github.com/fish546-2018/Jenny-Snailfish/blob/master/progress/InstallingStacks.md)*  
**Week 6**: Demultiplex data, and trim barcodes. Step completed but results didn't save due to Virtual Machine running out of space
**Week 7**: Resolve issues with virtual machine running out of space, by deleting all non-essential things and removing shared folders. Accessed data from external harddrive instead
**Week 8**: Rerun `process_radtags` now that VM issues are resolved.

**Upcoming**  
**Week 9**: Filter by read count. Determine number of loci to keep, compare across whole data set as well as within smaller groups. Construct a maximum likelihood phylogeny using Ape package in R.  
**Week 10**: Compare my phylogeny to already existing COI and RADSeq phylogenies. Was I able to recreate the same phylogeny as my coauthors using Stacks? If not, how do they differ? Does it affect crucial interpretation of the phylogeny?


## Repository Structure:
Detailed repository structure can be found [here]()

## Programs Used for Data Processing
- FastQC 
	- requires java
- Trim Galore
	- requires Cutadapt (which requires Microsoft C++)
- Stacks
- R (Ape package)
