{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This version of the Notebook details how to trip my data. I did not rerun FastQC on the untrimmed data so those chunks have been deleted. See Data-QT for notes on how FastQC was run on untrimmed data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/c/Users/FishLab3/Documents/GitHub/Jenny-Snailfish/notebooks\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " USAGE:\n",
      "\n",
      "trim_galore [options] <filename(s)>\n",
      "\n",
      "\n",
      "-h/--help               Print this help message and exits.\n",
      "\n",
      "-v/--version            Print the version information and exits.\n",
      "\n",
      "-q/--quality <INT>      Trim low-quality ends from reads in addition to adapter removal. For\n",
      "                        RRBS samples, quality trimming will be performed first, and adapter\n",
      "                        trimming is carried in a second round. Other files are quality and adapter\n",
      "                        trimmed in a single pass. The algorithm is the same as the one used by BWA\n",
      "                        (Subtract INT from all qualities; compute partial sums from all indices\n",
      "                        to the end of the sequence; cut sequence at the index at which the sum is\n",
      "                        minimal). Default Phred score: 20.\n",
      "\n",
      "--phred33               Instructs Cutadapt to use ASCII+33 quality scores as Phred scores\n",
      "                        (Sanger/Illumina 1.9+ encoding) for quality trimming. Default: ON.\n",
      "\n",
      "--phred64               Instructs Cutadapt to use ASCII+64 quality scores as Phred scores\n",
      "                        (Illumina 1.5 encoding) for quality trimming.\n",
      "\n",
      "--fastqc                Run FastQC in the default mode on the FastQ file once trimming is complete.\n",
      "\n",
      "--fastqc_args \"<ARGS>\"  Passes extra arguments to FastQC. If more than one argument is to be passed\n",
      "                        to FastQC they must be in the form \"arg1 arg2 etc.\". An example would be:\n",
      "                        --fastqc_args \"--nogroup --outdir /home/\". Passing extra arguments will\n",
      "                        automatically invoke FastQC, so --fastqc does not have to be specified\n",
      "                        separately.\n",
      "\n",
      "-a/--adapter <STRING>   Adapter sequence to be trimmed. If not specified explicitly, Trim Galore will\n",
      "                        try to auto-detect whether the Illumina universal, Nextera transposase or Illumina\n",
      "                        small RNA adapter sequence was used. Also see '--illumina', '--nextera' and\n",
      "                        '--small_rna'. If no adapter can be detected within the first 1 million sequences\n",
      "                        of the first file specified Trim Galore defaults to '--illumina'. A single base\n",
      "                        may also be given as e.g. -a A{10}, to be expanded to -a AAAAAAAAAA.\n",
      "\n",
      "-a2/--adapter2 <STRING> Optional adapter sequence to be trimmed off read 2 of paired-end files. This\n",
      "                        option requires '--paired' to be specified as well. If the libraries to be trimmed\n",
      "                        are smallRNA then a2 will be set to the Illumina small RNA 5' adapter automatically\n",
      "                        (GATCGTCGGACT). A single base may also be given as e.g. -a2 A{10}, to be expanded\n",
      "                        to -a2 AAAAAAAAAA.\n",
      "\n",
      "--illumina              Adapter sequence to be trimmed is the first 13bp of the Illumina universal adapter\n",
      "                        'AGATCGGAAGAGC' instead of the default auto-detection of adapter sequence.\n",
      "\n",
      "--nextera               Adapter sequence to be trimmed is the first 12bp of the Nextera adapter\n",
      "                        'CTGTCTCTTATA' instead of the default auto-detection of adapter sequence.\n",
      "\n",
      "--small_rna             Adapter sequence to be trimmed is the first 12bp of the Illumina Small RNA 3' Adapter\n",
      "                        'TGGAATTCTCGG' instead of the default auto-detection of adapter sequence. Selecting\n",
      "                        to trim smallRNA adapters will also lower the --length value to 18bp. If the smallRNA\n",
      "                        libraries are paired-end then a2 will be set to the Illumina small RNA 5' adapter\n",
      "                        automatically (GATCGTCGGACT) unless -a 2 had been defined explicitly.\n",
      "\n",
      "--max_length <INT>      Discard reads that are longer than <INT> bp after trimming. This is only advised for\n",
      "                        smallRNA sequencing to remove non-small RNA sequences.\n",
      "\n",
      "\n",
      "--stringency <INT>      Overlap with adapter sequence required to trim a sequence. Defaults to a\n",
      "                        very stringent setting of 1, i.e. even a single bp of overlapping sequence\n",
      "                        will be trimmed off from the 3' end of any read.\n",
      "\n",
      "-e <ERROR RATE>         Maximum allowed error rate (no. of errors divided by the length of the matching\n",
      "                        region) (default: 0.1)\n",
      "\n",
      "--gzip                  Compress the output file with GZIP. If the input files are GZIP-compressed\n",
      "                        the output files will automatically be GZIP compressed as well. As of v0.2.8 the\n",
      "                        compression will take place on the fly.\n",
      "\n",
      "--dont_gzip             Output files won't be compressed with GZIP. This option overrides --gzip.\n",
      "\n",
      "--length <INT>          Discard reads that became shorter than length INT because of either\n",
      "                        quality or adapter trimming. A value of '0' effectively disables\n",
      "                        this behaviour. Default: 20 bp.\n",
      "\n",
      "                        For paired-end files, both reads of a read-pair need to be longer than\n",
      "                        <INT> bp to be printed out to validated paired-end files (see option --paired).\n",
      "                        If only one read became too short there is the possibility of keeping such\n",
      "                        unpaired single-end reads (see --retain_unpaired). Default pair-cutoff: 20 bp.\n",
      "\n",
      "--max_n COUNT           The total number of Ns (as integer) a read may contain before it will be removed altogether.\n",
      "                        In a paired-end setting, either read exceeding this limit will result in the entire\n",
      "                        pair being removed from the trimmed output files.\n",
      "\n",
      "--trim-n                Removes Ns from either side of the read. This option does currently not work in RRBS mode.\n",
      "\n",
      "-o/--output_dir <DIR>   If specified all output will be written to this directory instead of the current\n",
      "                        directory.\n",
      "\n",
      "--no_report_file        If specified no report file will be generated.\n",
      "\n",
      "--suppress_warn         If specified any output to STDOUT or STDERR will be suppressed.\n",
      "\n",
      "--clip_R1 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 1 (or single-end\n",
      "                        reads). This may be useful if the qualities were very poor, or if there is some\n",
      "                        sort of unwanted bias at the 5' end. Default: OFF.\n",
      "\n",
      "--clip_R2 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 2 (paired-end reads\n",
      "                        only). This may be useful if the qualities were very poor, or if there is some sort\n",
      "                        of unwanted bias at the 5' end. For paired-end BS-Seq, it is recommended to remove\n",
      "                        the first few bp because the end-repair reaction may introduce a bias towards low\n",
      "                        methylation. Please refer to the M-bias plot section in the Bismark User Guide for\n",
      "                        some examples. Default: OFF.\n",
      "\n",
      "--three_prime_clip_R1 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 1 (or single-end\n",
      "                        reads) AFTER adapter/quality trimming has been performed. This may remove some unwanted\n",
      "                        bias from the 3' end that is not directly related to adapter sequence or basecall quality.\n",
      "                        Default: OFF.\n",
      "\n",
      "--three_prime_clip_R2 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 2 AFTER\n",
      "                        adapter/quality trimming has been performed. This may remove some unwanted bias from\n",
      "                        the 3' end that is not directly related to adapter sequence or basecall quality.\n",
      "                        Default: OFF.\n",
      "\n",
      "                     \n",
      "\n",
      "--path_to_cutadapt </path/to/cutadapt>     You may use this option to specify a path to the Cutadapt executable,\n",
      "                        e.g. /my/home/cutadapt-1.7.1/bin/cutadapt. Else it is assumed that Cutadapt is in\n",
      "                        the PATH.\n",
      "\n",
      "SPECIFIC TRIMMING - without adapter/quality trimming\n",
      "\n",
      "--hardtrim5 <int>       Instead of performing adapter-/quality trimming, this option will simply hard-trim sequences\n",
      "                        to <int> bp from the 5'-end. Once hard-trimming of files is complete, Trim Galore will exit.\n",
      "                        Hard-trimmed output files will end in .<int>.fq(.gz).\n",
      "  \n",
      "--clock                 In this mode, reads are trimmed in a specific way that is currently used for the Mouse\n",
      "                        Epigenetic Clock (see here: Multi-tissue DNA methylation age predictor in mouse, Stubbs et al.,\n",
      "                        Genome Biology, 2017 18:68 https://doi.org/10.1186/s13059-017-1203-5). Following this, Trim Galore\n",
      "                        will exit.\n",
      "\n",
      "                        In it's current implementation, the dual-UMI RRBS reads come in the following format:\n",
      "\n",
      "                        Read 1  5' UUUUUUUU CAGTA FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF TACTG UUUUUUUU 3'\n",
      "\t\t\tRead 2  3' UUUUUUUU GTCAT FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF ATGAC UUUUUUUU 5'\n",
      "\n",
      "                        Where UUUUUUUU is a random 8-mer unique molecular identifier (UMI), CAGTA is a constant region,\n",
      "                        and FFFFFFF... is the actual RRBS-Fragment to be sequenced. The UMIs for Read 1 (R1) and\n",
      "                        Read 2 (R2), as well as the fixed sequences (F1 or F2), are written into the read ID and \n",
      "                        removed from the actual sequence. Here is an example:\n",
      "                   \n",
      "                        R1: @HWI-D00436:407:CCAETANXX:1:1101:4105:1905 1:N:0: CGATGTTT\n",
      "                            ATCTAGTTCAGTACGGTGTTTTCGAATTAGAAAAATATGTATAGAGGAAATAGATATAAAGGCGTATTCGTTATTG\n",
      "                        R2: @HWI-D00436:407:CCAETANXX:1:1101:4105:1905 3:N:0: CGATGTTT\n",
      "                            CAATTTTGCAGTACAAAAATAATACCTCCTCTATTTATCCAAAATCACAAAAAACCACCCACTTAACTTTCCCTAA\n",
      "\n",
      "                        R1: @HWI-D00436:407:CCAETANXX:1:1101:4105:1905 1:N:0: CGATGTTT:R1:ATCTAGTT:R2:CAATTTTG:F1:CAGT:F2:CAGT\n",
      "                                         CGGTGTTTTCGAATTAGAAAAATATGTATAGAGGAAATAGATATAAAGGCGTATTCGTTATTG\n",
      "                        R2: @HWI-D00436:407:CCAETANXX:1:1101:4105:1905 3:N:0: CGATGTTT:R1:ATCTAGTT:R2:CAATTTTG:F1:CAGT:F2:CAGT\n",
      "                                         CAAAAATAATACCTCCTCTATTTATCCAAAATCACAAAAAACCACCCACTTAACTTTCCCTAA\n",
      "\n",
      "                        Following clock trimming, the resulting files (.clock_UMI.R1.fq(.gz) and .clock_UMI.R2.fq(.gz))\n",
      "                        should be adapter- and quality trimmed with Trim Galore as usual. In addition, reads need to be trimmed\n",
      "                        by 15bp from their 3' end to get rid of potential UMI and fixed sequences. The command is:\n",
      "\n",
      "                        trim_galore --paired --three_prime_clip_R1 15 --three_prime_clip_R2 15 *.clock_UMI.R1.fq.gz *.clock_UMI.R2.fq.gz\n",
      "\n",
      "                        Following this, reads should be aligned with Bismark and deduplicated with UmiBam \n",
      "                        in '--dual_index' mode (see here: https://github.com/FelixKrueger/Umi-Grinder). UmiBam recognises\n",
      "                        the UMIs within this pattern: R1:(ATCTAGTT):R2:(CAATTTTG): as (UMI R1) and (UMI R2).\n",
      "\n",
      "RRBS-specific options (MspI digested material):\n",
      "\n",
      "--rrbs                  Specifies that the input file was an MspI digested RRBS sample (recognition\n",
      "                        site: CCGG). Single-end or Read 1 sequences (paired-end) which were adapter-trimmed\n",
      "                        will have a further 2 bp removed from their 3' end. Sequences which were merely\n",
      "                        trimmed because of poor quality will not be shortened further. Read 2 of paired-end\n",
      "                        libraries will in addition have the first 2 bp removed from the 5' end (by setting \n",
      "                        '--clip_r2 2'). This is to avoid using artificial methylation calls from the filled-in\n",
      "                        cytosine positions close to the 3' MspI site in sequenced fragments. \n",
      "                        This option is not recommended for users of the NuGEN ovation RRBS System 1-16\n",
      "                        kit (see below).\n",
      "\n",
      "--non_directional       Selecting this option for non-directional RRBS libraries will screen\n",
      "                        quality-trimmed sequences for 'CAA' or 'CGA' at the start of the read\n",
      "                        and, if found, removes the first two basepairs. Like with the option\n",
      "                        '--rrbs' this avoids using cytosine positions that were filled-in\n",
      "                        during the end-repair step. '--non_directional' requires '--rrbs' to\n",
      "                        be specified as well. Note that this option does not set '--clip_r2 2' in\n",
      "                        paired-end mode.\n",
      "\n",
      "--keep                  Keep the quality trimmed intermediate file. Default: off, which means\n",
      "                        the temporary file is being deleted after adapter trimming. Only has\n",
      "                        an effect for RRBS samples since other FastQ files are not trimmed\n",
      "                        for poor qualities separately.\n",
      "\n",
      "\n",
      "Note for RRBS using the NuGEN Ovation RRBS System 1-16 kit:\n",
      "\n",
      "Owing to the fact that the NuGEN Ovation kit attaches a varying number of nucleotides (0-3) after each MspI\n",
      "site Trim Galore should be run WITHOUT the option --rrbs. This trimming is accomplished in a subsequent \n",
      "diversity trimming step afterwards (see their manual).\n",
      "\n",
      "\n",
      "\n",
      "Note for RRBS using MseI:\n",
      "\n",
      "If your DNA material was digested with MseI (recognition motif: TTAA) instead of MspI it is NOT necessary\n",
      "to specify --rrbs or --non_directional since virtually all reads should start with the sequence\n",
      "'TAA', and this holds true for both directional and non-directional libraries. As the end-repair of 'TAA'\n",
      "restricted sites does not involve any cytosines it does not need to be treated especially. Instead, simply\n",
      "run Trim Galore! in the standard (i.e. non-RRBS) mode.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Paired-end specific options:\n",
      "\n",
      "--paired                This option performs length trimming of quality/adapter/RRBS trimmed reads for\n",
      "                        paired-end files. To pass the validation test, both sequences of a sequence pair\n",
      "                        are required to have a certain minimum length which is governed by the option\n",
      "                        --length (see above). If only one read passes this length threshold the\n",
      "                        other read can be rescued (see option --retain_unpaired). Using this option lets\n",
      "                        you discard too short read pairs without disturbing the sequence-by-sequence order\n",
      "                        of FastQ files which is required by many aligners.\n",
      "\n",
      "                        Trim Galore! expects paired-end files to be supplied in a pairwise fashion, e.g.\n",
      "                        file1_1.fq file1_2.fq SRR2_1.fq.gz SRR2_2.fq.gz ... .\n",
      "\n",
      "-t/--trim1              Trims 1 bp off every read from its 3' end. This may be needed for FastQ files that\n",
      "                        are to be aligned as paired-end data with Bowtie. This is because Bowtie (1) regards\n",
      "                        alignments like this:\n",
      "\n",
      "                          R1 --------------------------->     or this:    ----------------------->  R1\n",
      "                          R2 <---------------------------                       <-----------------  R2\n",
      "\n",
      "                        as invalid (whenever a start/end coordinate is contained within the other read).\n",
      "                        NOTE: If you are planning to use Bowtie2, BWA etc. you don't need to specify this option.\n",
      "\n",
      "--retain_unpaired       If only one of the two paired-end reads became too short, the longer\n",
      "                        read will be written to either '.unpaired_1.fq' or '.unpaired_2.fq'\n",
      "                        output files. The length cutoff for unpaired single-end reads is\n",
      "                        governed by the parameters -r1/--length_1 and -r2/--length_2. Default: OFF.\n",
      "\n",
      "-r1/--length_1 <INT>    Unpaired single-end read length cutoff needed for read 1 to be written to\n",
      "                        '.unpaired_1.fq' output file. These reads may be mapped in single-end mode.\n",
      "                        Default: 35 bp.\n",
      "\n",
      "-r2/--length_2 <INT>    Unpaired single-end read length cutoff needed for read 2 to be written to\n",
      "                        '.unpaired_2.fq' output file. These reads may be mapped in single-end mode.\n",
      "                        Default: 35 bp.\n",
      "\n",
      "\n",
      "Last modified on 28 June 2018.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/c/Users/FishLab3/Bioinfo-progs/TrimGalore-0.5.0/trim_galore -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.18\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ../data/1078_S1_L001_R1_001.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t7266\tAGATCGGAAGAGC\t1000000\t0.73\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 7266). Second best hit was Nextera (count: 2)\n",
      "\n",
      "Writing report to '../analyses/1078_S1_L001_R1_001.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ../data/1078_S1_L001_R1_001.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\r\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "\n",
      "Writing final adapter and quality trimmed output to 1078_S1_L001_R1_001_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ../data/1078_S1_L001_R1_001.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "70000000 sequences processed\n",
      "80000000 sequences processed\n",
      "90000000 sequences processed\n",
      "100000000 sequences processed\n",
      "110000000 sequences processed\n",
      "120000000 sequences processed\n",
      "130000000 sequences processed\n",
      "140000000 sequences processed\n",
      "150000000 sequences processed\n",
      "160000000 sequences processed\n",
      "170000000 sequences processed\n",
      "180000000 sequences processed\n",
      "190000000 sequences processed\n",
      "200000000 sequences processed\n",
      "210000000 sequences processed\n",
      "220000000 sequences processed\n",
      "230000000 sequences processed\n",
      "240000000 sequences processed\n",
      "250000000 sequences processed\n",
      "260000000 sequences processed\n",
      "270000000 sequences processed\n",
      "280000000 sequences processed\n",
      "290000000 sequences processed\n",
      "300000000 sequences processed\n",
      "310000000 sequences processed\n",
      "320000000 sequences processed\n",
      "330000000 sequences processed\n",
      "340000000 sequences processed\n",
      "350000000 sequences processed\n",
      "360000000 sequences processed\n",
      "370000000 sequences processed\n",
      "380000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.7.0\r\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ../data/1078_S1_L001_R1_001.fastq\r\n",
      "Processing reads on 1 core in single-end mode ...\r\n",
      "Finished in 7739.81 s (20 us/read; 3.01 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total reads processed:             388,390,826\r\n",
      "Reads with adapters:               122,931,149 (31.7%)\r\n",
      "Reads written (passing filters):   388,390,826 (100.0%)\r\n",
      "\r\n",
      "Total basepairs processed: 58,647,014,726 bp\r\n",
      "Quality-trimmed:           2,057,106,867 bp (3.5%)\r\n",
      "Total written (filtered):  56,243,187,950 bp (95.9%)\r\n",
      "\r\n",
      "=== Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 122931149 times.\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-9 bp: 0; 10-13 bp: 1\r\n",
      "\r\n",
      "Bases preceding removed adapters:\r\n",
      "  A: 34.1%\r\n",
      "  C: 30.4%\r\n",
      "  G: 19.7%\r\n",
      "  T: 15.9%\r\n",
      "  none/other: 0.0%\r\n",
      "\r\n",
      "Overview of removed sequences\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "1\t87778718\t97097706.5\t0\t87778718\r\n",
      "2\t23602869\t24274426.6\t0\t23602869\r\n",
      "3\t6338219\t6068606.7\t0\t6338219\r\n",
      "4\t1142973\t1517151.7\t0\t1142973\r\n",
      "5\t287006\t379287.9\t0\t287006\r\n",
      "6\t128586\t94822.0\t0\t128586\r\n",
      "7\t123232\t23705.5\t0\t123232\r\n",
      "8\t72247\t5926.4\t0\t72247\r\n",
      "9\t103446\t1481.6\t0\t93205 10241\r\n",
      "10\t86882\t370.4\t1\t75077 11805\r\n",
      "11\t68298\t92.6\t1\t58412 9886\r\n",
      "12\t85379\t23.1\t1\t78058 7321\r\n",
      "13\t62061\t5.8\t1\t55973 6088\r\n",
      "14\t82295\t5.8\t1\t72930 9365\r\n",
      "15\t112328\t5.8\t1\t99442 12886\r\n",
      "16\t28468\t5.8\t1\t25641 2827\r\n",
      "17\t58859\t5.8\t1\t55244 3615\r\n",
      "18\t50866\t5.8\t1\t47273 3593\r\n",
      "19\t48173\t5.8\t1\t44809 3364\r\n",
      "20\t52571\t5.8\t1\t49006 3565\r\n",
      "21\t54994\t5.8\t1\t50578 4416\r\n",
      "22\t41286\t5.8\t1\t37984 3302\r\n",
      "23\t45688\t5.8\t1\t41464 4224\r\n",
      "24\t58821\t5.8\t1\t49364 9457\r\n",
      "25\t36744\t5.8\t1\t34046 2698\r\n",
      "26\t44086\t5.8\t1\t40887 3199\r\n",
      "27\t34468\t5.8\t1\t32479 1989\r\n",
      "28\t36320\t5.8\t1\t33727 2593\r\n",
      "29\t35551\t5.8\t1\t32672 2879\r\n",
      "30\t38615\t5.8\t1\t35346 3269\r\n",
      "31\t44636\t5.8\t1\t40497 4139\r\n",
      "32\t30120\t5.8\t1\t28147 1973\r\n",
      "33\t23667\t5.8\t1\t21980 1687\r\n",
      "34\t22757\t5.8\t1\t20966 1791\r\n",
      "35\t26578\t5.8\t1\t24570 2008\r\n",
      "36\t30979\t5.8\t1\t28543 2436\r\n",
      "37\t37982\t5.8\t1\t34878 3104\r\n",
      "38\t43329\t5.8\t1\t40513 2816\r\n",
      "39\t10335\t5.8\t1\t9336 999\r\n",
      "40\t10743\t5.8\t1\t9909 834\r\n",
      "41\t11578\t5.8\t1\t10622 956\r\n",
      "42\t19320\t5.8\t1\t18176 1144\r\n",
      "43\t19492\t5.8\t1\t18196 1296\r\n",
      "44\t17196\t5.8\t1\t16142 1054\r\n",
      "45\t14778\t5.8\t1\t13814 964\r\n",
      "46\t17057\t5.8\t1\t16170 887\r\n",
      "47\t15763\t5.8\t1\t14560 1203\r\n",
      "48\t16776\t5.8\t1\t15422 1354\r\n",
      "49\t19534\t5.8\t1\t17924 1610\r\n",
      "50\t24487\t5.8\t1\t22661 1826\r\n",
      "51\t11193\t5.8\t1\t10138 1055\r\n",
      "52\t12971\t5.8\t1\t11968 1003\r\n",
      "53\t11951\t5.8\t1\t10309 1642\r\n",
      "54\t13182\t5.8\t1\t12267 915\r\n",
      "55\t12938\t5.8\t1\t11981 957\r\n",
      "56\t11471\t5.8\t1\t10774 697\r\n",
      "57\t10255\t5.8\t1\t9422 833\r\n",
      "58\t15013\t5.8\t1\t13600 1413\r\n",
      "59\t12277\t5.8\t1\t11120 1157\r\n",
      "60\t12087\t5.8\t1\t11039 1048\r\n",
      "61\t13482\t5.8\t1\t12018 1464\r\n",
      "62\t14497\t5.8\t1\t8834 5663\r\n",
      "63\t11834\t5.8\t1\t10447 1387\r\n",
      "64\t11904\t5.8\t1\t10535 1369\r\n",
      "65\t13265\t5.8\t1\t10882 2383\r\n",
      "66\t15294\t5.8\t1\t10512 4782\r\n",
      "67\t12549\t5.8\t1\t11385 1164\r\n",
      "68\t13144\t5.8\t1\t11771 1373\r\n",
      "69\t16076\t5.8\t1\t14466 1610\r\n",
      "70\t25281\t5.8\t1\t20834 4447\r\n",
      "71\t183971\t5.8\t1\t176598 7373\r\n",
      "72\t200195\t5.8\t1\t193360 6835\r\n",
      "73\t164213\t5.8\t1\t158404 5809\r\n",
      "74\t126966\t5.8\t1\t122254 4712\r\n",
      "75\t88874\t5.8\t1\t85549 3325\r\n",
      "76\t61010\t5.8\t1\t57863 3147\r\n",
      "77\t43699\t5.8\t1\t41490 2209\r\n",
      "78\t35068\t5.8\t1\t31607 3461\r\n",
      "79\t28826\t5.8\t1\t25709 3117\r\n",
      "80\t23493\t5.8\t1\t21021 2472\r\n",
      "81\t21317\t5.8\t1\t17949 3368\r\n",
      "82\t17021\t5.8\t1\t15610 1411\r\n",
      "83\t15827\t5.8\t1\t14375 1452\r\n",
      "84\t28834\t5.8\t1\t27718 1116\r\n",
      "85\t28237\t5.8\t1\t27444 793\r\n",
      "86\t22535\t5.8\t1\t21781 754\r\n",
      "87\t15491\t5.8\t1\t14793 698\r\n",
      "88\t10970\t5.8\t1\t10228 742\r\n",
      "89\t8434\t5.8\t1\t7863 571\r\n",
      "90\t6852\t5.8\t1\t6279 573\r\n",
      "91\t6106\t5.8\t1\t5536 570\r\n",
      "92\t5530\t5.8\t1\t5021 509\r\n",
      "93\t4799\t5.8\t1\t4250 549\r\n",
      "94\t4505\t5.8\t1\t3892 613\r\n",
      "95\t4097\t5.8\t1\t3655 442\r\n",
      "96\t4656\t5.8\t1\t4142 514\r\n",
      "97\t3650\t5.8\t1\t3086 564\r\n",
      "98\t3721\t5.8\t1\t2917 804\r\n",
      "99\t3345\t5.8\t1\t2731 614\r\n",
      "100\t4344\t5.8\t1\t2569 1775\r\n",
      "101\t3019\t5.8\t1\t2571 448\r\n",
      "102\t3065\t5.8\t1\t2640 425\r\n",
      "103\t2769\t5.8\t1\t2360 409\r\n",
      "104\t2622\t5.8\t1\t2263 359\r\n",
      "105\t2464\t5.8\t1\t2103 361\r\n",
      "106\t2367\t5.8\t1\t1999 368\r\n",
      "107\t2478\t5.8\t1\t2027 451\r\n",
      "108\t2471\t5.8\t1\t1906 565\r\n",
      "109\t2437\t5.8\t1\t1802 635\r\n",
      "110\t4401\t5.8\t1\t1749 2652\r\n",
      "111\t2610\t5.8\t1\t1837 773\r\n",
      "112\t6437\t5.8\t1\t1727 4710\r\n",
      "113\t1931\t5.8\t1\t1659 272\r\n",
      "114\t1905\t5.8\t1\t1647 258\r\n",
      "115\t1743\t5.8\t1\t1497 246\r\n",
      "116\t1776\t5.8\t1\t1504 272\r\n",
      "117\t1775\t5.8\t1\t1484 291\r\n",
      "118\t2752\t5.8\t1\t1421 1331\r\n",
      "119\t1645\t5.8\t1\t1395 250\r\n",
      "120\t1838\t5.8\t1\t1451 387\r\n",
      "121\t2273\t5.8\t1\t1592 681\r\n",
      "122\t2769\t5.8\t1\t2523 246\r\n",
      "123\t7895\t5.8\t1\t7651 244\r\n",
      "124\t5902\t5.8\t1\t5529 373\r\n",
      "125\t5967\t5.8\t1\t5750 217\r\n",
      "126\t4508\t5.8\t1\t4243 265\r\n",
      "127\t4974\t5.8\t1\t4743 231\r\n",
      "128\t5442\t5.8\t1\t5130 312\r\n",
      "129\t4849\t5.8\t1\t4268 581\r\n",
      "130\t2710\t5.8\t1\t2350 360\r\n",
      "131\t3075\t5.8\t1\t2646 429\r\n",
      "132\t3515\t5.8\t1\t2779 736\r\n",
      "133\t3629\t5.8\t1\t2660 969\r\n",
      "134\t8392\t5.8\t1\t4780 3612\r\n",
      "135\t33790\t5.8\t1\t21428 12362\r\n",
      "136\t2773\t5.8\t1\t2471 302\r\n",
      "137\t1816\t5.8\t1\t1752 64\r\n",
      "138\t1759\t5.8\t1\t1700 59\r\n",
      "139\t1194\t5.8\t1\t1141 53\r\n",
      "140\t1374\t5.8\t1\t1320 54\r\n",
      "141\t1884\t5.8\t1\t1811 73\r\n",
      "142\t2292\t5.8\t1\t2185 107\r\n",
      "143\t2828\t5.8\t1\t2707 121\r\n",
      "144\t8997\t5.8\t1\t8600 397\r\n",
      "145\t93067\t5.8\t1\t89672 3395\r\n",
      "146\t3933\t5.8\t1\t3824 109\r\n",
      "147\t2263\t5.8\t1\t2202 61\r\n",
      "148\t2047\t5.8\t1\t1955 92\r\n",
      "149\t1470\t5.8\t1\t1378 92\r\n",
      "150\t2381\t5.8\t1\t2156 225\r\n",
      "151\t4470\t5.8\t1\t2525 1945\r\n",
      "\r\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ../data/1078_S1_L001_R1_001.fastq\n",
      "=============================================\n",
      "388390826 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t971936 (0.3%)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/c/Users/FishLab3/Bioinfo-progs/TrimGalore-0.5.0/trim_galore -o ../analyses/ ../data/1078_S1_L001_R1_001.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then ran FastQC on trimmed data for comparison to original FastQC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for 1078_S1_L001_R1_001_trimmed.fq\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 5% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 10% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 15% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 20% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 25% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 30% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 35% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 40% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 45% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 50% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 55% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 60% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 65% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 70% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 75% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 80% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 85% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 90% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n",
      "Approx 95% complete for 1078_S1_L001_R1_001_trimmed.fq\r\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/c/Users/FishLab3/Bioinfo-progs/FastQC/fastqc \\\n",
    "-j /c/Users/FishLab3/Bioinfo-progs/Java/jre1.8.0_191/bin/java.exe \\\n",
    "../data/1078_S1_L001_R1_001_trimmed.fq \\\n",
    "-o ../data/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Based on comparison of FastQC from raw data and FastQC from trimmed data, it appears I need to use more options in Trim Galore to trim more bases off."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "/c/Users/FishLab3/Bioinfo-progs/TrimGalore-0.5.0/trim_galore -o ../analyses/ ../data/1078_S1_L001_R1_001_trimmed.fq"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
