{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is the notebook I used for running `sstacks` on my 22 specimens in my catalog."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "sstacks 2.2\n",
      "sstacks -P dir -M popmap [-p n_threads]\n",
      "sstacks -c catalog_path -s sample_path [-s sample_path ...] -o path [-p n_threads]\n",
      "  -P,--in_path: path to the directory containing Stacks files.\n",
      "  -M,--popmap: path to a population map file from which to take sample names.\n",
      "  -s,--sample: filename prefix from which to load sample loci.\n",
      "  -c,--catalog: path to the catalog.\n",
      "  -p,--threads: enable parallel execution with num_threads threads.\n",
      "  -o,--out_path: output path to write results.\n",
      "  -x: don't verify haplotype of matching locus.\n",
      "\n",
      "Gapped assembly options:\n",
      "  --disable_gapped: disable gapped alignments between stacks (default: enable gapped alignments).\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "sstacks -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Searching for matches by sequence identity...\n",
      "  Parsing ./stacks/catalog.tags.tsv.gz\n",
      "  Parsing ./stacks/catalog.snps.tsv.gz\n",
      "  Parsing ./stacks/catalog.alleles.tsv.gz\n",
      "Populating kmer dictionary for exact matches...done.\n",
      "\n",
      "Processing sample 'stacks/Lflorae666-cap' [1 of 1]\n",
      "  Parsing stacks/Lflorae666-cap.tags.tsv.gz\n",
      "  Parsing stacks/Lflorae666-cap.snps.tsv.gz\n",
      "  Parsing stacks/Lflorae666-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "45783 sample loci compared against the catalog containing 293517 loci.\n",
      "  28678 matching loci, 8 contained no verified haplotypes.\n",
      "  8 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  34849 total haplotypes examined from matching loci, 34812 verified.\n",
      "Outputing to file ./Lflorae666-cap.matches.tsv.gz\n",
      "\n",
      "sstacks is done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "sstacks -c ./stacks/ -s stacks/Lflorae666-cap -p 15 --disable_gapped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# sstacks version 2.2; generated on 2018-12-04 08:55:53\n",
      "272514\t25\t1\tconsensus\t26\t126M\n",
      "272515\t25\t3\tconsensus\t38\t134M\n",
      "272516\t25\t7\tconsensus\t73\t125M\n",
      "272517\t25\t9\tconsensus\t60\t127M\n",
      "272518\t25\t10\tG\t12\t128M\n",
      "272518\t25\t10\tT\t7\t128M\n",
      "1811\t25\t11\tCCCCTCACCTCTATGGGTACTAAAGA\t39\t125M\n",
      "272519\t25\t12\tC\t19\t131M\n",
      "272519\t25\t12\tT\t24\t131M\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "zcat ./Lflorae666-cap.matches.tsv.gz | head -n 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running into issues with gapped alignments so ran `sstacks` with `--disable_gapped` option and it worked."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir stacks/sstacks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Searching for matches by sequence identity...\n",
      "  Parsing ./stacks/catalog.tags.tsv.gz\n",
      "  Parsing ./stacks/catalog.snps.tsv.gz\n",
      "  Parsing ./stacks/catalog.alleles.tsv.gz\n",
      "Populating kmer dictionary for exact matches...done.\n",
      "\n",
      "Processing sample './stacks/Ebritti822-cap' [1 of 44]\n",
      "  Parsing ./stacks/Ebritti822-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ebritti822-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ebritti822-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41391 sample loci compared against the catalog containing 332639 loci.\n",
      "  32882 matching loci, 25 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  25 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  38122 total haplotypes examined from matching loci, 38097 verified.\n",
      "Outputing to file ./Ebritti822-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Lferxi9829-cap' [2 of 44]\n",
      "  Parsing ./stacks/Lferxi9829-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lferxi9829-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lferxi9829-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "39162 sample loci compared against the catalog containing 332639 loci.\n",
      "  14434 matching loci, 126 contained no verified haplotypes.\n",
      "  5 loci matched more than one catalog locus and were excluded.\n",
      "  121 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  16800 total haplotypes examined from matching loci, 16667 verified.\n",
      "Outputing to file ./Lferxi9829-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Pulochir76' [3 of 44]\n",
      "  Parsing ./stacks/Pulochir76.tags.tsv.gz\n",
      "  Parsing ./stacks/Pulochir76.snps.tsv.gz\n",
      "  Parsing ./stacks/Pulochir76.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "42705 sample loci compared against the catalog containing 332639 loci.\n",
      "  19353 matching loci, 439 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  439 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  22162 total haplotypes examined from matching loci, 21721 verified.\n",
      "Outputing to file ./Pulochir76.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ajordani86-cap' [4 of 44]\n",
      "  Parsing ./stacks/Ajordani86-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordani86-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordani86-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "44024 sample loci compared against the catalog containing 332639 loci.\n",
      "  14700 matching loci, 277 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  277 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  16627 total haplotypes examined from matching loci, 16349 verified.\n",
      "Outputing to file ./Ajordani86-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Lrutteri56' [5 of 44]\n",
      "  Parsing ./stacks/Lrutteri56.tags.tsv.gz\n",
      "  Parsing ./stacks/Lrutteri56.snps.tsv.gz\n",
      "  Parsing ./stacks/Lrutteri56.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "34286 sample loci compared against the catalog containing 332639 loci.\n",
      "  22179 matching loci, 13 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  13 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  24005 total haplotypes examined from matching loci, 23992 verified.\n",
      "Outputing to file ./Lrutteri56.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ecaudatus2-cap' [6 of 44]\n",
      "  Parsing ./stacks/Ecaudatus2-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus2-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus2-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41743 sample loci compared against the catalog containing 332639 loci.\n",
      "  14565 matching loci, 199 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  199 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  16576 total haplotypes examined from matching loci, 16373 verified.\n",
      "Outputing to file ./Ecaudatus2-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Pdactyl08' [7 of 44]\n",
      "  Parsing ./stacks/Pdactyl08.tags.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl08.snps.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl08.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "11654 sample loci compared against the catalog containing 332639 loci.\n",
      "  6691 matching loci, 16 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  16 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  6830 total haplotypes examined from matching loci, 6813 verified.\n",
      "Outputing to file ./Pdactyl08.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ccandidus-cap' [8 of 44]\n",
      "  Parsing ./stacks/Ccandidus-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccandidus-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccandidus-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "45411 sample loci compared against the catalog containing 332639 loci.\n",
      "  15167 matching loci, 252 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  252 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  17949 total haplotypes examined from matching loci, 17694 verified.\n",
      "Outputing to file ./Ccandidus-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ecaudatus3' [9 of 44]\n",
      "  Parsing ./stacks/Ecaudatus3.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus3.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus3.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "51638 sample loci compared against the catalog containing 332639 loci.\n",
      "  23537 matching loci, 227 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  227 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  25700 total haplotypes examined from matching loci, 25472 verified.\n",
      "Outputing to file ./Ecaudatus3.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ccfcypselur-cap' [10 of 44]\n",
      "  Parsing ./stacks/Ccfcypselur-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccfcypselur-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccfcypselur-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "53523 sample loci compared against the catalog containing 332639 loci.\n",
      "  24138 matching loci, 164 contained no verified haplotypes.\n",
      "  3 loci matched more than one catalog locus and were excluded.\n",
      "  161 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  26218 total haplotypes examined from matching loci, 26053 verified.\n",
      "Outputing to file ./Ccfcypselur-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Pdactyl51-cap' [11 of 44]\n",
      "  Parsing ./stacks/Pdactyl51-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl51-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl51-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "42421 sample loci compared against the catalog containing 332639 loci.\n",
      "  14465 matching loci, 230 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  230 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  17508 total haplotypes examined from matching loci, 17278 verified.\n",
      "Outputing to file ./Pdactyl51-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ccfmelanu2-cap' [12 of 44]\n",
      "  Parsing ./stacks/Ccfmelanu2-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu2-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu2-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43733 sample loci compared against the catalog containing 332639 loci.\n",
      "  15283 matching loci, 81 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  81 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  16202 total haplotypes examined from matching loci, 16119 verified.\n",
      "Outputing to file ./Ccfmelanu2-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Lnanus2476-cap' [13 of 44]\n",
      "  Parsing ./stacks/Lnanus2476-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lnanus2476-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lnanus2476-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "46159 sample loci compared against the catalog containing 332639 loci.\n",
      "  23683 matching loci, 375 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  375 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  27223 total haplotypes examined from matching loci, 26848 verified.\n",
      "Outputing to file ./Lnanus2476-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Crystall87' [14 of 44]\n",
      "  Parsing ./stacks/Crystall87.tags.tsv.gz\n",
      "  Parsing ./stacks/Crystall87.snps.tsv.gz\n",
      "  Parsing ./stacks/Crystall87.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "11921 sample loci compared against the catalog containing 332639 loci.\n",
      "  5435 matching loci, 11 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  11 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  5623 total haplotypes examined from matching loci, 5612 verified.\n",
      "Outputing to file ./Crystall87.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cscottae04-cap' [15 of 44]\n",
      "  Parsing ./stacks/Cscottae04-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cscottae04-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cscottae04-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40348 sample loci compared against the catalog containing 332639 loci.\n",
      "  14430 matching loci, 71 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  71 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  15848 total haplotypes examined from matching loci, 15777 verified.\n",
      "Outputing to file ./Cscottae04-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ajordani89-cap' [16 of 44]\n",
      "  Parsing ./stacks/Ajordani89-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordani89-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordani89-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "43462 sample loci compared against the catalog containing 332639 loci.\n",
      "  14949 matching loci, 234 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  234 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  17387 total haplotypes examined from matching loci, 17153 verified.\n",
      "Outputing to file ./Ajordani89-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ajordan521-cap' [17 of 44]\n",
      "  Parsing ./stacks/Ajordan521-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordan521-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordan521-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "49588 sample loci compared against the catalog containing 332639 loci.\n",
      "  20481 matching loci, 210 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  210 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  23108 total haplotypes examined from matching loci, 22898 verified.\n",
      "Outputing to file ./Ajordan521-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ccomus970-cap' [18 of 44]\n",
      "  Parsing ./stacks/Ccomus970-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccomus970-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccomus970-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "46654 sample loci compared against the catalog containing 332639 loci.\n",
      "  18260 matching loci, 192 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  192 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  21513 total haplotypes examined from matching loci, 21320 verified.\n",
      "Outputing to file ./Ccomus970-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Eorbis821-cap' [19 of 44]\n",
      "  Parsing ./stacks/Eorbis821-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Eorbis821-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Eorbis821-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40388 sample loci compared against the catalog containing 332639 loci.\n",
      "  31417 matching loci, 1 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  1 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  36394 total haplotypes examined from matching loci, 36393 verified.\n",
      "Outputing to file ./Eorbis821-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ecaudatus4-cap' [20 of 44]\n",
      "  Parsing ./stacks/Ecaudatus4-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus4-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus4-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "41545 sample loci compared against the catalog containing 332639 loci.\n",
      "  15104 matching loci, 133 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  133 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  17628 total haplotypes examined from matching loci, 17495 verified.\n",
      "Outputing to file ./Ecaudatus4-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cmelanu88' [21 of 44]\n",
      "  Parsing ./stacks/Cmelanu88.tags.tsv.gz\n",
      "  Parsing ./stacks/Cmelanu88.snps.tsv.gz\n",
      "  Parsing ./stacks/Cmelanu88.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "36841 sample loci compared against the catalog containing 332639 loci.\n",
      "  14410 matching loci, 27 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  27 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  15109 total haplotypes examined from matching loci, 15081 verified.\n",
      "Outputing to file ./Cmelanu88.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Lflorae666-cap' [22 of 44]\n",
      "  Parsing ./stacks/Lflorae666-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lflorae666-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lflorae666-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "45783 sample loci compared against the catalog containing 332639 loci.\n",
      "  28981 matching loci, 40 contained no verified haplotypes.\n",
      "  8 loci matched more than one catalog locus and were excluded.\n",
      "  32 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  35152 total haplotypes examined from matching loci, 35083 verified.\n",
      "Outputing to file ./Lflorae666-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ajordan522-cap' [23 of 44]\n",
      "  Parsing ./stacks/Ajordan522-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordan522-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordan522-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "52728 sample loci compared against the catalog containing 332639 loci.\n",
      "  10804 matching loci, 417 contained no verified haplotypes.\n",
      "  2 loci matched more than one catalog locus and were excluded.\n",
      "  415 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  12614 total haplotypes examined from matching loci, 12181 verified.\n",
      "Outputing to file ./Ajordan522-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Pptychoman-cap' [24 of 44]\n",
      "  Parsing ./stacks/Pptychoman-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Pptychoman-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Pptychoman-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "44254 sample loci compared against the catalog containing 332639 loci.\n",
      "  3273 matching loci, 408 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  408 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3288 total haplotypes examined from matching loci, 2880 verified.\n",
      "Outputing to file ./Pptychoman-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ecaudatus1-cap' [25 of 44]\n",
      "  Parsing ./stacks/Ecaudatus1-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus1-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus1-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "50702 sample loci compared against the catalog containing 332639 loci.\n",
      "  11718 matching loci, 390 contained no verified haplotypes.\n",
      "  2 loci matched more than one catalog locus and were excluded.\n",
      "  388 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  13057 total haplotypes examined from matching loci, 12648 verified.\n",
      "Outputing to file ./Ecaudatus1-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cgilberti81-cap' [26 of 44]\n",
      "  Parsing ./stacks/Cgilberti81-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cgilberti81-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cgilberti81-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "42627 sample loci compared against the catalog containing 332639 loci.\n",
      "  5006 matching loci, 733 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  733 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  5089 total haplotypes examined from matching loci, 4351 verified.\n",
      "Outputing to file ./Cgilberti81-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Npelagicus-cap' [27 of 44]\n",
      "  Parsing ./stacks/Npelagicus-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Npelagicus-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Npelagicus-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "46453 sample loci compared against the catalog containing 332639 loci.\n",
      "  595 matching loci, 90 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  90 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  597 total haplotypes examined from matching loci, 507 verified.\n",
      "Outputing to file ./Npelagicus-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ccandidu2' [28 of 44]\n",
      "  Parsing ./stacks/Ccandidu2.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccandidu2.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccandidu2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "20267 sample loci compared against the catalog containing 332639 loci.\n",
      "  3192 matching loci, 221 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  221 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3239 total haplotypes examined from matching loci, 3018 verified.\n",
      "Outputing to file ./Ccandidu2.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ccomus971-cap' [29 of 44]\n",
      "  Parsing ./stacks/Ccomus971-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccomus971-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccomus971-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "47886 sample loci compared against the catalog containing 332639 loci.\n",
      "  11614 matching loci, 431 contained no verified haplotypes.\n",
      "  1 loci matched more than one catalog locus and were excluded.\n",
      "  430 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  13762 total haplotypes examined from matching loci, 13312 verified.\n",
      "Outputing to file ./Ccomus971-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Lgibbus176-cap' [30 of 44]\n",
      "  Parsing ./stacks/Lgibbus176-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus176-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus176-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "45132 sample loci compared against the catalog containing 332639 loci.\n",
      "  3550 matching loci, 491 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  491 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3581 total haplotypes examined from matching loci, 3089 verified.\n",
      "Outputing to file ./Lgibbus176-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cfaunus841' [31 of 44]\n",
      "  Parsing ./stacks/Cfaunus841.tags.tsv.gz\n",
      "  Parsing ./stacks/Cfaunus841.snps.tsv.gz\n",
      "  Parsing ./stacks/Cfaunus841.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "33056 sample loci compared against the catalog containing 332639 loci.\n",
      "  5312 matching loci, 548 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  548 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  5447 total haplotypes examined from matching loci, 4892 verified.\n",
      "Outputing to file ./Cfaunus841.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Aungak696-cap' [32 of 44]\n",
      "  Parsing ./stacks/Aungak696-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Aungak696-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Aungak696-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "44419 sample loci compared against the catalog containing 332639 loci.\n",
      "  7813 matching loci, 912 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  912 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  8347 total haplotypes examined from matching loci, 7425 verified.\n",
      "Outputing to file ./Aungak696-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ccfmelanu1-cap' [33 of 44]\n",
      "  Parsing ./stacks/Ccfmelanu1-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu1-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu1-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "40988 sample loci compared against the catalog containing 332639 loci.\n",
      "  8617 matching loci, 285 contained no verified haplotypes.\n",
      "  1 loci matched more than one catalog locus and were excluded.\n",
      "  284 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  8849 total haplotypes examined from matching loci, 8552 verified.\n",
      "Outputing to file ./Ccfmelanu1-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cstauffe96-cap' [34 of 44]\n",
      "  Parsing ./stacks/Cstauffe96-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cstauffe96-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cstauffe96-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "44631 sample loci compared against the catalog containing 332639 loci.\n",
      "  3877 matching loci, 403 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  403 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  3924 total haplotypes examined from matching loci, 3519 verified.\n",
      "Outputing to file ./Cstauffe96-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Eorbis461-cap' [35 of 44]\n",
      "  Parsing ./stacks/Eorbis461-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Eorbis461-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Eorbis461-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "46153 sample loci compared against the catalog containing 332639 loci.\n",
      "  17892 matching loci, 3524 contained no verified haplotypes.\n",
      "  2 loci matched more than one catalog locus and were excluded.\n",
      "  3522 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  18424 total haplotypes examined from matching loci, 14881 verified.\n",
      "Outputing to file ./Eorbis461-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cfurcellu98-cap' [36 of 44]\n",
      "  Parsing ./stacks/Cfurcellu98-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cfurcellu98-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cfurcellu98-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "44410 sample loci compared against the catalog containing 332639 loci.\n",
      "  6213 matching loci, 602 contained no verified haplotypes.\n",
      "  2 loci matched more than one catalog locus and were excluded.\n",
      "  600 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  6352 total haplotypes examined from matching loci, 5737 verified.\n",
      "Outputing to file ./Cfurcellu98-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cscottae02-cap' [37 of 44]\n",
      "  Parsing ./stacks/Cscottae02-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cscottae02-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cscottae02-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "46274 sample loci compared against the catalog containing 332639 loci.\n",
      "  8395 matching loci, 511 contained no verified haplotypes.\n",
      "  1 loci matched more than one catalog locus and were excluded.\n",
      "  510 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  8684 total haplotypes examined from matching loci, 8167 verified.\n",
      "Outputing to file ./Cscottae02-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cphasma07-cap' [38 of 44]\n",
      "  Parsing ./stacks/Cphasma07-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cphasma07-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cphasma07-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "44497 sample loci compared against the catalog containing 332639 loci.\n",
      "  5699 matching loci, 684 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  684 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  5866 total haplotypes examined from matching loci, 5180 verified.\n",
      "Outputing to file ./Cphasma07-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Crystalli2-cap' [39 of 44]\n",
      "  Parsing ./stacks/Crystalli2-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Crystalli2-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Crystalli2-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "59132 sample loci compared against the catalog containing 332639 loci.\n",
      "  4890 matching loci, 387 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  387 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  4907 total haplotypes examined from matching loci, 4519 verified.\n",
      "Outputing to file ./Crystalli2-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Crystalli1' [40 of 44]\n",
      "  Parsing ./stacks/Crystalli1.tags.tsv.gz\n",
      "  Parsing ./stacks/Crystalli1.snps.tsv.gz\n",
      "  Parsing ./stacks/Crystalli1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "58438 sample loci compared against the catalog containing 332639 loci.\n",
      "  5162 matching loci, 450 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  450 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  5184 total haplotypes examined from matching loci, 4734 verified.\n",
      "Outputing to file ./Crystalli1.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Cphasma61-cap' [41 of 44]\n",
      "  Parsing ./stacks/Cphasma61-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cphasma61-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cphasma61-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "50139 sample loci compared against the catalog containing 332639 loci.\n",
      "  5630 matching loci, 664 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  664 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  5796 total haplotypes examined from matching loci, 5128 verified.\n",
      "Outputing to file ./Cphasma61-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Lgibbus043-cap' [42 of 44]\n",
      "  Parsing ./stacks/Lgibbus043-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus043-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus043-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "50198 sample loci compared against the catalog containing 332639 loci.\n",
      "  1782 matching loci, 213 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  213 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  1786 total haplotypes examined from matching loci, 1573 verified.\n",
      "Outputing to file ./Lgibbus043-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Ppenicillu-cap' [43 of 44]\n",
      "  Parsing ./stacks/Ppenicillu-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ppenicillu-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ppenicillu-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "49524 sample loci compared against the catalog containing 332639 loci.\n",
      "  2713 matching loci, 446 contained no verified haplotypes.\n",
      "  0 loci matched more than one catalog locus and were excluded.\n",
      "  446 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  2742 total haplotypes examined from matching loci, 2296 verified.\n",
      "Outputing to file ./Ppenicillu-cap.matches.tsv.gz\n",
      "\n",
      "Processing sample './stacks/Rattenuatu-cap' [44 of 44]\n",
      "  Parsing ./stacks/Rattenuatu-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Rattenuatu-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Rattenuatu-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "45138 sample loci compared against the catalog containing 332639 loci.\n",
      "  45137 matching loci, 12 contained no verified haplotypes.\n",
      "  12 loci matched more than one catalog locus and were excluded.\n",
      "  0 loci contained SNPs unaccounted for in the catalog and were excluded.\n",
      "  58049 total haplotypes examined from matching loci, 58024 verified.\n",
      "Outputing to file ./Rattenuatu-cap.matches.tsv.gz\n",
      "\n",
      "sstacks is done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bash run_sstacks.sh"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
