{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running cstacks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This the notebook I used for running cstacks to create my catalog. Done after running ustacks and stacking my reads into loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/media/jenny/3TB_NFTS_ext/data\r\n"
     ]
    }
   ],
   "source": [
    "! pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cstacks 2.2\n",
      "cstacks -P in_dir -M popmap [-n num_mismatches] [-p num_threads]\n",
      "cstacks -s sample1_path [-s sample2_path ...] -o path [-n num_mismatches] [-p num_threads]\n",
      "\n",
      "  -P,--in_path: path to the directory containing Stacks files.\n",
      "  -M,--popmap: path to a population map file.\n",
      "  -n: number of mismatches allowed between sample loci when build the catalog (default 1; suggested: set to ustacks -M).\n",
      "  -p,--threads: enable parallel execution with num_threads threads.\n",
      "  -s: sample prefix from which to load loci into the catalog.\n",
      "  -o,--outpath: output path to write results.\n",
      "  -c,--catalog <path>: add to an existing catalog.\n",
      "\n",
      "Gapped assembly options:\n",
      "  --max_gaps: number of gaps allowed between stacks before merging (default: 2).\n",
      "  --min_aln_len: minimum length of aligned sequence in a gapped alignment (default: 0.80).\n",
      "  --disable_gapped: disable gapped alignments between stacks (default: use gapped alignments).\n",
      "\n",
      "Advanced options:\n",
      "  --k_len <len>: specify k-mer size for matching between between catalog loci (automatically calculated by default).\n",
      "  --report_mmatches: report query loci that match more than one catalog locus.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cstacks -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I kept running into issues where it would fail at sample 22. So I tried running it in pieces, but still failed."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`cstacks` always fails at the code after Lflorae666 when searching for gapped alignments. Try running something from after there on it's own and adding to catalog and disabling gapped alignments."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cstacks parameters selected:\n",
      "  Loci matched based on sequence identity.\n",
      "  Number of mismatches allowed between stacks: 10\n",
      "  Gapped alignments: disabled\n",
      "Constructing catalog from 1 samples.\n",
      "\n",
      "Initializing existing catalog...\n",
      "  Parsing ./stacks/catalog.tags.tsv.gz\n",
      "  Parsing ./stacks/catalog.snps.tsv.gz\n",
      "  Parsing ./stacks/catalog.alleles.tsv.gz\n",
      "\n",
      "Processing sample ./stacks/Rattenuatu-cap [1 of 1]\n",
      "  Parsing ./stacks/Rattenuatu-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Rattenuatu-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Rattenuatu-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  293517 loci in the catalog, 4532335 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  6016 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  39130 loci were newly added to the catalog.\n",
      "  8 loci matched more than one catalog locus, linking them.\n",
      "    16 linked catalog loci were merged into 8 loci.\n",
      "\n",
      "Writing catalog in directory './stacks/'.\n",
      "Final catalog contains 332639 loci.\n",
      "cstacks is done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cstacks -o ./stacks/ --catalog ./stacks/catalog -s ./stacks/Rattenuatu-cap -n 10 -p 15 --disable_gapped"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It worked. So now try running the whole thing with `disable_gapped`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir stacks/totcat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cstacks parameters selected:\n",
      "  Loci matched based on sequence identity.\n",
      "  Number of mismatches allowed between stacks: 10\n",
      "  Gapped alignments: disabled\n",
      "Constructing catalog from 44 samples.\n",
      "\n",
      "Initializing new catalog...\n",
      "  Parsing ./stacks/Ebritti822-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ebritti822-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ebritti822-cap.alleles.tsv.gz\n",
      "  41391 loci were newly added to the catalog.\n",
      "\n",
      "Processing sample ./stacks/Lferxi9829-cap [2 of 44]\n",
      "  Parsing ./stacks/Lferxi9829-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lferxi9829-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lferxi9829-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  41391 loci in the catalog, 2189003 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  2099 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  37065 loci were newly added to the catalog.\n",
      "  2 loci matched more than one catalog locus, linking them.\n",
      "    4 linked catalog loci were merged into 2 loci.\n",
      "\n",
      "Processing sample ./stacks/Pulochir76 [3 of 44]\n",
      "  Parsing ./stacks/Pulochir76.tags.tsv.gz\n",
      "  Parsing ./stacks/Pulochir76.snps.tsv.gz\n",
      "  Parsing ./stacks/Pulochir76.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  78454 loci in the catalog, 2856688 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  10261 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  32461 loci were newly added to the catalog.\n",
      "  17 loci matched more than one catalog locus, linking them.\n",
      "    26 linked catalog loci were merged into 12 loci.\n",
      "\n",
      "Processing sample ./stacks/Ajordani86-cap [4 of 44]\n",
      "  Parsing ./stacks/Ajordani86-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordani86-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordani86-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  110901 loci in the catalog, 3141129 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  17592 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  26465 loci were newly added to the catalog.\n",
      "  33 loci matched more than one catalog locus, linking them.\n",
      "    66 linked catalog loci were merged into 33 loci.\n",
      "\n",
      "Processing sample ./stacks/Lrutteri56 [5 of 44]\n",
      "  Parsing ./stacks/Lrutteri56.tags.tsv.gz\n",
      "  Parsing ./stacks/Lrutteri56.snps.tsv.gz\n",
      "  Parsing ./stacks/Lrutteri56.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  137333 loci in the catalog, 3289150 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  6257 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  28035 loci were newly added to the catalog.\n",
      "  6 loci matched more than one catalog locus, linking them.\n",
      "    12 linked catalog loci were merged into 6 loci.\n",
      "\n",
      "Processing sample ./stacks/Ecaudatus2-cap [6 of 44]\n",
      "  Parsing ./stacks/Ecaudatus2-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus2-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus2-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  165362 loci in the catalog, 3453312 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  17541 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  24246 loci were newly added to the catalog.\n",
      "  43 loci matched more than one catalog locus, linking them.\n",
      "    86 linked catalog loci were merged into 42 loci.\n",
      "\n",
      "Processing sample ./stacks/Pdactyl08 [7 of 44]\n",
      "  Parsing ./stacks/Pdactyl08.tags.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl08.snps.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl08.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  189564 loci in the catalog, 3528077 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  4155 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  7507 loci were newly added to the catalog.\n",
      "  8 loci matched more than one catalog locus, linking them.\n",
      "    16 linked catalog loci were merged into 8 loci.\n",
      "\n",
      "Processing sample ./stacks/Ccandidus-cap [8 of 44]\n",
      "  Parsing ./stacks/Ccandidus-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccandidus-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccandidus-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  197063 loci in the catalog, 3564995 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  21804 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  23696 loci were newly added to the catalog.\n",
      "  89 loci matched more than one catalog locus, linking them.\n",
      "    174 linked catalog loci were merged into 86 loci.\n",
      "\n",
      "Processing sample ./stacks/Ecaudatus3 [9 of 44]\n",
      "  Parsing ./stacks/Ecaudatus3.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus3.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus3.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  220671 loci in the catalog, 3624692 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  30478 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  21324 loci were newly added to the catalog.\n",
      "  162 loci matched more than one catalog locus, linking them.\n",
      "    319 linked catalog loci were merged into 158 loci.\n",
      "\n",
      "Processing sample ./stacks/Ccfcypselur-cap [10 of 44]\n",
      "  Parsing ./stacks/Ccfcypselur-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccfcypselur-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccfcypselur-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  241834 loci in the catalog, 3690298 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  23301 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  30278 loci were newly added to the catalog.\n",
      "  56 loci matched more than one catalog locus, linking them.\n",
      "    112 linked catalog loci were merged into 56 loci.\n",
      "\n",
      "Processing sample ./stacks/Pdactyl51-cap [11 of 44]\n",
      "  Parsing ./stacks/Pdactyl51-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl51-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Pdactyl51-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  272056 loci in the catalog, 3767616 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  23297 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  19266 loci were newly added to the catalog.\n",
      "  141 loci matched more than one catalog locus, linking them.\n",
      "    281 linked catalog loci were merged into 139 loci.\n",
      "\n",
      "Processing sample ./stacks/Ccfmelanu2-cap [12 of 44]\n",
      "  Parsing ./stacks/Ccfmelanu2-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu2-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu2-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  291180 loci in the catalog, 3794038 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  28942 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  14848 loci were newly added to the catalog.\n",
      "  55 loci matched more than one catalog locus, linking them.\n",
      "    110 linked catalog loci were merged into 54 loci.\n",
      "\n",
      "Processing sample ./stacks/Lnanus2476-cap [13 of 44]\n",
      "  Parsing ./stacks/Lnanus2476-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lnanus2476-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lnanus2476-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  305972 loci in the catalog, 3815927 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  19560 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  26677 loci were newly added to the catalog.\n",
      "  78 loci matched more than one catalog locus, linking them.\n",
      "    156 linked catalog loci were merged into 78 loci.\n",
      "\n",
      "Processing sample ./stacks/Crystall87 [14 of 44]\n",
      "  Parsing ./stacks/Crystall87.tags.tsv.gz\n",
      "  Parsing ./stacks/Crystall87.snps.tsv.gz\n",
      "  Parsing ./stacks/Crystall87.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  332571 loci in the catalog, 3844471 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  5604 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  6321 loci were newly added to the catalog.\n",
      "  4 loci matched more than one catalog locus, linking them.\n",
      "    8 linked catalog loci were merged into 4 loci.\n",
      "\n",
      "Processing sample ./stacks/Cscottae04-cap [15 of 44]\n",
      "  Parsing ./stacks/Cscottae04-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cscottae04-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cscottae04-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  338888 loci in the catalog, 3854112 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  24725 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  15697 loci were newly added to the catalog.\n",
      "  74 loci matched more than one catalog locus, linking them.\n",
      "    145 linked catalog loci were merged into 72 loci.\n",
      "\n",
      "Processing sample ./stacks/Ajordani89-cap [16 of 44]\n",
      "  Parsing ./stacks/Ajordani89-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordani89-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordani89-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  354512 loci in the catalog, 3868883 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  28276 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  15381 loci were newly added to the catalog.\n",
      "  193 loci matched more than one catalog locus, linking them.\n",
      "    388 linked catalog loci were merged into 193 loci.\n",
      "\n",
      "Processing sample ./stacks/Ajordan521-cap [17 of 44]\n",
      "  Parsing ./stacks/Ajordan521-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordan521-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordan521-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  369698 loci in the catalog, 3881016 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  31134 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  18751 loci were newly added to the catalog.\n",
      "  294 loci matched more than one catalog locus, linking them.\n",
      "    590 linked catalog loci were merged into 293 loci.\n",
      "\n",
      "Processing sample ./stacks/Ccomus970-cap [18 of 44]\n",
      "  Parsing ./stacks/Ccomus970-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccomus970-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccomus970-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  388152 loci in the catalog, 3890460 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  27871 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  18991 loci were newly added to the catalog.\n",
      "  207 loci matched more than one catalog locus, linking them.\n",
      "    415 linked catalog loci were merged into 207 loci.\n",
      "\n",
      "Processing sample ./stacks/Eorbis821-cap [19 of 44]\n",
      "  Parsing ./stacks/Eorbis821-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Eorbis821-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Eorbis821-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  406935 loci in the catalog, 3906208 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  31453 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  8995 loci were newly added to the catalog.\n",
      "  60 loci matched more than one catalog locus, linking them.\n",
      "    119 linked catalog loci were merged into 59 loci.\n",
      "\n",
      "Processing sample ./stacks/Ecaudatus4-cap [20 of 44]\n",
      "  Parsing ./stacks/Ecaudatus4-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus4-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus4-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  415870 loci in the catalog, 3910720 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  33036 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  8754 loci were newly added to the catalog.\n",
      "  244 loci matched more than one catalog locus, linking them.\n",
      "    488 linked catalog loci were merged into 243 loci.\n",
      "\n",
      "Processing sample ./stacks/Cmelanu88 [21 of 44]\n",
      "  Parsing ./stacks/Cmelanu88.tags.tsv.gz\n",
      "  Parsing ./stacks/Cmelanu88.snps.tsv.gz\n",
      "  Parsing ./stacks/Cmelanu88.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  424379 loci in the catalog, 3914670 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  26823 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  10064 loci were newly added to the catalog.\n",
      "  46 loci matched more than one catalog locus, linking them.\n",
      "    92 linked catalog loci were merged into 46 loci.\n",
      "\n",
      "Processing sample ./stacks/Lflorae666-cap [22 of 44]\n",
      "  Parsing ./stacks/Lflorae666-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lflorae666-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lflorae666-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  434397 loci in the catalog, 3923577 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  12852 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  32966 loci were newly added to the catalog.\n",
      "  35 loci matched more than one catalog locus, linking them.\n",
      "    68 linked catalog loci were merged into 33 loci.\n",
      "\n",
      "Processing sample ./stacks/Ajordan522-cap [23 of 44]\n",
      "  Parsing ./stacks/Ajordan522-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ajordan522-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ajordan522-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  467328 loci in the catalog, 3959247 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  37865 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  14975 loci were newly added to the catalog.\n",
      "  112 loci matched more than one catalog locus, linking them.\n",
      "    224 linked catalog loci were merged into 112 loci.\n",
      "\n",
      "Processing sample ./stacks/Pptychoman-cap [24 of 44]\n",
      "  Parsing ./stacks/Pptychoman-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Pptychoman-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Pptychoman-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  482191 loci in the catalog, 3967587 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  23603 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  20737 loci were newly added to the catalog.\n",
      "  82 loci matched more than one catalog locus, linking them.\n",
      "    166 linked catalog loci were merged into 81 loci.\n",
      "\n",
      "Processing sample ./stacks/Ecaudatus1-cap [25 of 44]\n",
      "  Parsing ./stacks/Ecaudatus1-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus1-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ecaudatus1-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  502843 loci in the catalog, 3980126 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  38904 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  11872 loci were newly added to the catalog.\n",
      "  74 loci matched more than one catalog locus, linking them.\n",
      "    148 linked catalog loci were merged into 74 loci.\n",
      "\n",
      "Processing sample ./stacks/Cgilberti81-cap [26 of 44]\n",
      "  Parsing ./stacks/Cgilberti81-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cgilberti81-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cgilberti81-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  514641 loci in the catalog, 3984445 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  28619 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  14216 loci were newly added to the catalog.\n",
      "  208 loci matched more than one catalog locus, linking them.\n",
      "    416 linked catalog loci were merged into 208 loci.\n",
      "\n",
      "Processing sample ./stacks/Npelagicus-cap [27 of 44]\n",
      "  Parsing ./stacks/Npelagicus-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Npelagicus-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Npelagicus-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  528649 loci in the catalog, 3992004 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  10697 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  35799 loci were newly added to the catalog.\n",
      "  43 loci matched more than one catalog locus, linking them.\n",
      "    86 linked catalog loci were merged into 43 loci.\n",
      "\n",
      "Processing sample ./stacks/Ccandidu2 [28 of 44]\n",
      "  Parsing ./stacks/Ccandidu2.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccandidu2.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccandidu2.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  564405 loci in the catalog, 4017421 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  16530 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  3790 loci were newly added to the catalog.\n",
      "  53 loci matched more than one catalog locus, linking them.\n",
      "    106 linked catalog loci were merged into 53 loci.\n",
      "\n",
      "Processing sample ./stacks/Ccomus971-cap [29 of 44]\n",
      "  Parsing ./stacks/Ccomus971-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccomus971-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccomus971-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  568142 loci in the catalog, 4018997 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  38177 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  9790 loci were newly added to the catalog.\n",
      "  80 loci matched more than one catalog locus, linking them.\n",
      "    156 linked catalog loci were merged into 76 loci.\n",
      "\n",
      "Processing sample ./stacks/Lgibbus176-cap [30 of 44]\n",
      "  Parsing ./stacks/Lgibbus176-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus176-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus176-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  577852 loci in the catalog, 4022363 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  20429 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  24807 loci were newly added to the catalog.\n",
      "  104 loci matched more than one catalog locus, linking them.\n",
      "    208 linked catalog loci were merged into 104 loci.\n",
      "\n",
      "Processing sample ./stacks/Cfaunus841 [31 of 44]\n",
      "  Parsing ./stacks/Cfaunus841.tags.tsv.gz\n",
      "  Parsing ./stacks/Cfaunus841.snps.tsv.gz\n",
      "  Parsing ./stacks/Cfaunus841.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  602555 loci in the catalog, 4034945 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  25960 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  7217 loci were newly added to the catalog.\n",
      "  121 loci matched more than one catalog locus, linking them.\n",
      "    241 linked catalog loci were merged into 120 loci.\n",
      "\n",
      "Processing sample ./stacks/Aungak696-cap [32 of 44]\n",
      "  Parsing ./stacks/Aungak696-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Aungak696-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Aungak696-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  609651 loci in the catalog, 4037833 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  34487 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  10250 loci were newly added to the catalog.\n",
      "  316 loci matched more than one catalog locus, linking them.\n",
      "    629 linked catalog loci were merged into 313 loci.\n",
      "\n",
      "Processing sample ./stacks/Ccfmelanu1-cap [33 of 44]\n",
      "  Parsing ./stacks/Ccfmelanu1-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu1-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ccfmelanu1-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  619585 loci in the catalog, 4041013 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  33219 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  7856 loci were newly added to the catalog.\n",
      "  85 loci matched more than one catalog locus, linking them.\n",
      "    171 linked catalog loci were merged into 84 loci.\n",
      "\n",
      "Processing sample ./stacks/Cstauffe96-cap [34 of 44]\n",
      "  Parsing ./stacks/Cstauffe96-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cstauffe96-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cstauffe96-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  627354 loci in the catalog, 4041897 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  25516 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  19216 loci were newly added to the catalog.\n",
      "  101 loci matched more than one catalog locus, linking them.\n",
      "    202 linked catalog loci were merged into 101 loci.\n",
      "\n",
      "Processing sample ./stacks/Eorbis461-cap [35 of 44]\n",
      "  Parsing ./stacks/Eorbis461-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Eorbis461-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Eorbis461-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  646469 loci in the catalog, 4047921 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  30615 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  15636 loci were newly added to the catalog.\n",
      "  98 loci matched more than one catalog locus, linking them.\n",
      "    195 linked catalog loci were merged into 97 loci.\n",
      "\n",
      "Processing sample ./stacks/Cfurcellu98-cap [36 of 44]\n",
      "  Parsing ./stacks/Cfurcellu98-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cfurcellu98-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cfurcellu98-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  662007 loci in the catalog, 4054038 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  31660 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  12959 loci were newly added to the catalog.\n",
      "  206 loci matched more than one catalog locus, linking them.\n",
      "    413 linked catalog loci were merged into 205 loci.\n",
      "\n",
      "Processing sample ./stacks/Cscottae02-cap [37 of 44]\n",
      "  Parsing ./stacks/Cscottae02-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cscottae02-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cscottae02-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  674758 loci in the catalog, 4057104 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  33135 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  13286 loci were newly added to the catalog.\n",
      "  145 loci matched more than one catalog locus, linking them.\n",
      "    292 linked catalog loci were merged into 145 loci.\n",
      "\n",
      "Processing sample ./stacks/Cphasma07-cap [38 of 44]\n",
      "  Parsing ./stacks/Cphasma07-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cphasma07-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cphasma07-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  687897 loci in the catalog, 4061079 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  32136 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  12613 loci were newly added to the catalog.\n",
      "  249 loci matched more than one catalog locus, linking them.\n",
      "    501 linked catalog loci were merged into 249 loci.\n",
      "\n",
      "Processing sample ./stacks/Crystalli2-cap [39 of 44]\n",
      "  Parsing ./stacks/Crystalli2-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Crystalli2-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Crystalli2-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  700258 loci in the catalog, 4064847 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  26075 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  33113 loci were newly added to the catalog.\n",
      "  54 loci matched more than one catalog locus, linking them.\n",
      "    110 linked catalog loci were merged into 54 loci.\n",
      "\n",
      "Processing sample ./stacks/Crystalli1 [40 of 44]\n",
      "  Parsing ./stacks/Crystalli1.tags.tsv.gz\n",
      "  Parsing ./stacks/Crystalli1.snps.tsv.gz\n",
      "  Parsing ./stacks/Crystalli1.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  733315 loci in the catalog, 4078372 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  35356 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  23132 loci were newly added to the catalog.\n",
      "  50 loci matched more than one catalog locus, linking them.\n",
      "    100 linked catalog loci were merged into 50 loci.\n",
      "\n",
      "Processing sample ./stacks/Cphasma61-cap [41 of 44]\n",
      "  Parsing ./stacks/Cphasma61-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Cphasma61-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Cphasma61-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  756397 loci in the catalog, 4085958 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  34164 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  16252 loci were newly added to the catalog.\n",
      "  272 loci matched more than one catalog locus, linking them.\n",
      "    549 linked catalog loci were merged into 272 loci.\n",
      "\n",
      "Processing sample ./stacks/Lgibbus043-cap [42 of 44]\n",
      "  Parsing ./stacks/Lgibbus043-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus043-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Lgibbus043-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  772372 loci in the catalog, 4089564 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  26384 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  23929 loci were newly added to the catalog.\n",
      "  113 loci matched more than one catalog locus, linking them.\n",
      "    227 linked catalog loci were merged into 112 loci.\n",
      "\n",
      "Processing sample ./stacks/Ppenicillu-cap [43 of 44]\n",
      "  Parsing ./stacks/Ppenicillu-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Ppenicillu-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Ppenicillu-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  796186 loci in the catalog, 4095280 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  22379 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  27298 loci were newly added to the catalog.\n",
      "  151 loci matched more than one catalog locus, linking them.\n",
      "    304 linked catalog loci were merged into 151 loci.\n",
      "\n",
      "Processing sample ./stacks/Rattenuatu-cap [44 of 44]\n",
      "  Parsing ./stacks/Rattenuatu-cap.tags.tsv.gz\n",
      "  Parsing ./stacks/Rattenuatu-cap.snps.tsv.gz\n",
      "  Parsing ./stacks/Rattenuatu-cap.alleles.tsv.gz\n",
      "Searching for sequence matches...\n",
      "  823331 loci in the catalog, 4100693 kmers in the catalog hash.\n",
      "Merging matches into catalog...\n",
      "  22704 loci were matched to a catalog locus.\n",
      "  0 loci were matched to a catalog locus using gapped alignments.\n",
      "  22581 loci were newly added to the catalog.\n",
      "  147 loci matched more than one catalog locus, linking them.\n",
      "    294 linked catalog loci were merged into 147 loci.\n",
      "\n",
      "Writing catalog in directory './stacks/totcat/'.\n",
      "Final catalog contains 845765 loci.\n",
      "cstacks is done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bash run_cstacks_jg.sh"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
